mbed/adapters/socketio_mbed_os5.cpp

```diff
@@ -224,8 +224,12 @@ static int send_queued_data(SOCKET_IO_INSTANCE* socket_io_instance)
             }
             else if (send_result < 0)
             {
-                indicate_error(socket_io_instance);
-                return -1;
+                if (send_result == NSAPI_ERROR_WOULD_BLOCK) {
+                    thread_sleep_for(10);
+                } else {
+                    indicate_error(socket_io_instance);
+                    return -1;
+                }
```

Comment:
```suggestion
                if (send_result != NSAPI_ERROR_WOULD_BLOCK) {
                    indicate_error(socket_io_instance);
                    return -1;
                }
```

Rather than `thread_sleep_for(10)`, maybe we can just carry on with NSAPI_ERROR_WOULD_BLOCK, similar to the function for retrieving: https://github.com/ARMmbed/mbed-client-for-azure/blob/16f9b31e41c1bb89b776418b6dc109b92b0d911c/mbed/adapters/socketio_mbed_os5.cpp#L178-L186

Reply:
@LDong-Arm OK. Updated as suggested.